draft-trammell-taps-post-sockets.md

```diff
@@ -818,12 +830,118 @@ func sayHelloWithAssociation(association Association) {
 }
 ~~~~~~~~
 
+## API Dynamics {#sec-dynamics}
+
+As Carriers provide the central entry point to Post, they are key to API
+dynamics. The lifecycle of a carrier is shown in {{fig-carrier-lc}}. Carriers
+are created by active openers by calling Initiate() given a Local and a Remote,
+and by passive openers by calling Listen() given a Local; the .Accept() method
+on the listener Carrier can then be used to create active carriers. By default,
+the underlying Association is automatically created and managed by the
+underlying API. This underlying Association can be accessed by the Carrier's
+.Association() method. Alternately, an association can be explicitly created
+using NewAssociation(), and a Carrier on the association may be accessed or
+initiated by calling the association's .GetCarrier() method.
+
+Once a Carrier has been created (via Initiate, NewSource, NewSink,
+Listen/Accept, or implicitly by calling Association.GetCarrier), it may be used
+to send and receive Messages. The existence of a Carrier does not imply the
+existence of an active Transient or associated transport-layer connection; these
+may be created when the carrier is, or may be deferred, depending on the network
+environment, configuration, and protocol stacks available.
+
+~~~~~~
+ Listen(local)   Initiate(local,remote)   NewSource(local,remote)
+      |                     |             or NewSink(local)
+  [ Carrier  ]              |                    |
+  [(listener)]              +--------------------+
+      |                     V                    
+  .Accept()-----------> [Carrier] -+----------> .Close()
+                         |     ^   |  close    [ Carrier  ]
+                         |     |   +- event -> [ (closed) ]
+                         |     |          
+              .Association()  .GetCarrier()
+                         |     |
+                         V     |
+                     [Association]   
+                           ^
+                           |
+              NewAssociation(local,remote)
+~~~~~~~
+{: #fig-carrier-lc title="Carrier and Association Life Cycle"}
+
+Access to more detailed information is possible through accessors on Carriers
+and Associations, following the relationships shown in {{fig-abstractions}}.
+Carriers allow their Transients to be accessed and enumerated, primarily for
+logging and debugging purposes; Associations likewise allow their Paths to be
+enumerated for access to cached path properties.
```

Comment:
Not completely sure what you mean by enumerated and why this is important here?

Reply:
There's a Path abstraction in the API, but in the current Internet architecture, the only thing you can really do with Paths is get information about them. So basically the idea here is you can get some reference to a collection of path objects ("enumerate" is admittedly a very functional-Pythonesque way to refer to this) to look at that information.

It's here because the point of the dynamics section (I think) is to talk about all the possible paths (heh) through the API. Agree that this needs to be made more concrete (something like "Association has a .Paths() method which allows the application to get information about the paths in use...".